File: backend/schemas/premium.py
from typing import Literal

from pydantic import BaseModel


class CheckoutRequest(BaseModel):
    plan_name: Literal["monthly", "yearly"]


class CheckoutResponse(BaseModel):
    checkout_url: str
    session_id: str
    plan_name: str


class WebhookResponse(BaseModel):
    received: bool

File: frontend/types/premium.ts
// Mirrors backend/schemas/premium.py 1:1 — do not add fields that don't exist there.

export interface CheckoutRequest {
  plan_name: "monthly" | "yearly"
}

export interface CheckoutResponse {
  checkout_url: string
  session_id:   string
  plan_name:    string
}

export interface WebhookResponse {
  received: boolean
}
